README


This archive contains the distribution Email-Address,
version 1.905:

  RFC 2822 Address Parsing and Creation

This software is copyright (c) 2004 by Casey West.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.019.



Changes
Release history for Email-Address

1.905     2014-06-17 22:55:00-04:00 America/New_York
        - additional change to avoid slowdown; this addresses CVE-2014-0477
          change provided by Bastian Blank <waldi@debian.org>

1.904     2014-06-14 00:21:21-04:00 America/New_York (TRIAL RELEASE)
        - avoid being fooled by an addr-like string in the phrase
        - avoid a slowdown by avoiding backtracking into the phrase

1.903     2014-04-17 21:02:14-04:00 America/New_York
        - correctly parenthesize false comment "0" (sigh)

1.902     2014-04-17 10:45:11-04:00 America/New_York
        - when formatting an address where phrase is empty but comment is not,
          do not include "" for the phrase; just omit it
        - when formatting and address where comment lacks enclosing parens, add
          them

1.901     2014-01-28 22:41:54-05:00 America/New_York
          further avoidance of stringifying to undef (thanks, this time, to
          David Golden)

1.900     2013-08-08 14:44:43 America/New_York
          avoid trying to stringify undef, and so avoid a warning (thanks, Alex
          Vandiver)

1.899     2013-08-02 10:53:19 America/New_York
          repackaged, pointed at new bugtracker

          fallback overloads are enabled, so "cmp" will now compare with string
          overloads... which still might not be what you expect

          code has been futzed with to rely on some 5.6-isms, rather than
          trying to be 5.005 compatible

          a warning has been added: calling ->parse in scalar context gets
          undefined behavior

1.898     2013-02-07
          remove dead link

1.897     2012-12-17
          skip mailboxes, rather than aborting the whole parse, when
          encountering non-ASCII characters (thanks, Ruslan Zakirov!)

          improve the behavior of escaping in phrases (thanks, Ruslan Zakirov!)

1.896     2012-07-31
          documentation improvements (thanks, Glenn Fowler!)

1.895     2012-01-15
          allow non-ASCII *only in phrase*

1.894     2012-01-14
          note that the use of the regex vars is not a great idea

1.893     2012-01-02
          reject any non-ascii content in strings

1.892     2010-09-02
          revert all behavior to 1.889

1.891     2010-08-30
          rework domainless address feature to work on perl5.8 (Alex Vandiver)

1.890     2010-08-22
          allow domainless addresses (if requested) (thanks, Alex Vandiver)

1.889     2007-12-19
          even if the phrase needed quoting, do not return quoted phrase from
          the phrase method

1.888     2007-04-01
          rt 16320 - collapse multiple whitespaces to prevent speed problem
          rt 24161 - forbid backslash in dot-atom

1.887     2007-04-01
          fix a test to prevent failure on 5.005

1.886     2007-03-01
          fix stupid use/require bug in test

1.885     2007-02-28
          do not quote the phrase if it's MIME encoded
          do not double quote the phrase
          thanks to MIYAGAWA and KEVINR for tests for the above!

1.884     2006-12-04
          fix phrase-quoting behavior by always quoting the phrase, if given
          this fixes bug 13387; thanks, Dave Rolsky!

1.883     2006-11-25
          resolve bug 23565, as_string didn't work on 5.8.0 (weird!)
          improve skip behavior for cache-cow.t with older Scalar::Util
          restore as_string
          issue deprecation warning if $STRINGIFY is changed

1.882     2006-11-22
          fix bug 21270, do not allow changes to objects to pollute the cache
          packaging improvements
          make test suite easier to manage
          improve honesty of documentation (we are not complete, (yet))

1.881     2006-11-20
          Basically: fixes a case where regex never returns.
          resolve bug 23187 by slighly diluting the solution to 22991, which is
            a more marginal problem than that of 23187, in practical terms

1.880     2006-11-11
          resolve bug 22991, support obs-phrase construct
          this creates a significant slowdown, sadly, but nothing to rival bug
            16320

1.871     2006-10-12
1.870     2006-08-10
          restore as_string
          issue deprecation warning if $STRINGIFY is changed

1.86      2006-07-21
          update docs
          tweak regex for approximate 50% performance boost
          ...but bug 16320 is still true: Email::Address can be VERY slow
          https://rt.cpan.org/Ticket/Display.html?id=16320


1.85      2006-07-07
          add disable_cache and enable_cache methods (bug 19163, thanks ADAMK)

1.80      2004-10-22
          With long mal-formed comment structures, backtracking was causing
          parse() to "hang." This was tracked down and fixed.  (Tim O'Brien
          reported this).

1.7       2004-10-12
          $comment was initialized twice, oops.
          Lowered the default nested comment level from five to two, cut regex
          in half.

1.6       2004-10-05
          Only define package variables if they haven't already been defined.

1.5       2004-10-05
          Implemented the remainder of the spec for CFWS. This was
          intentionally left out of previous versions because it increases the
          size of the regex ten fold. I was surprised to learn that the
          slowdown was only 33%, leaving Email-Address faster than its
          competition by a factor of two in the benchmarks I've created (which
          makes them subjective, I suppose).

1.3       2004-08-15
          name() was incorrectly fixing up a comment.

1.2       2004-06-02
          Added original() method.
          Allowed access to key regexes.

1.1       2004-05-26
          Initial version.


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.019.
use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "RFC 2822 Address Parsing and Creation",
  "AUTHOR" => "Casey West, Ricardo SIGNES <rjbs\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Email-Address",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Email::Address",
  "PREREQ_PM" => {
    "overload" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Encode" => 0,
    "Encode::MIME::Header" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "1.905",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Encode" => 0,
  "Encode::MIME::Header" => 0,
  "Test::More" => "0.96",
  "overload" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



